sdk/eventhubs/azure_messaging_eventhubs_checkpointstore_blob/src/checkpoint_store.rs

```diff
@@ -0,0 +1,324 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+//! Azure Blob Storage implementation of the Event Hubs checkpoint store.
+// cspell: ignore rfind sequencenumber ownerid
+use azure_core::{
+    http::{
+        headers::{ETAG, LAST_MODIFIED},
+        Etag, NoFormat, RequestContent, StatusCode,
+    },
+    time::parse_rfc7231,
+    Bytes, Result,
+};
+use azure_messaging_eventhubs::{
+    models::{Checkpoint, Ownership},
+    CheckpointStore,
+};
+use azure_storage_blob::{
+    models::{
+        BlobClientGetPropertiesResultHeaders, BlobClientSetMetadataOptions,
+        BlobContainerClientListBlobFlatSegmentOptions, BlockBlobClientUploadOptions,
+        BlockBlobClientUploadResultHeaders, ListBlobsIncludeItem,
+    },
+    BlobContainerClient,
+};
+use futures::TryStreamExt;
+use std::{collections::HashMap, fmt::Debug, sync::Arc};
+use time::OffsetDateTime;
+use tracing::{debug, warn};
+
+/// Azure Blob Storage implementation of the [`CheckpointStore`] trait.
+///
+/// This implementation stores checkpoint and ownership information in Azure Blob Storage,
+/// providing durable persistence for Event Hub event processing state.
+#[derive(Clone)]
+pub struct BlobCheckpointStore {
+    blob_container_client: Arc<BlobContainerClient>,
+}
+
+impl Debug for BlobCheckpointStore {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("BlobCheckpointStore").finish()
```

Comment:
Nit: I recommend you either derive `SafeDebug`, or use the following to show there are members, but not for thee:


```suggestion
        f.debug_struct("BlobCheckpointStore").finish_non_exhaustive()
```

Reply:
Or just `f.write_str("BlobCheckpointStore")` if you just want the name. What you have will result in `BlobCheckpointStore {}` which is a bit misleading.